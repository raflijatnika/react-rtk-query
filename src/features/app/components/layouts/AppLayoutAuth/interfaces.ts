// React
import { ReactNode } from 'react'

export interface IAppLayoutAuthProps {
  children?: ReactNode
  isLogin?: boolean
}
