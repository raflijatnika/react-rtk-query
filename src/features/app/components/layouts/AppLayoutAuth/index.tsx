// React
import { memo } from 'react'

// Components
import {
  StyledWrapper,
  StyledContainer,
  StyledFormContainer,
  StyledBanner
} from './components'
import { AppBaseLabel, AppRouteWrapper } from '@/features/app/components'

// i18n
import { useTranslation } from 'react-i18next'

// Interfaces
import { IAppLayoutAuthProps } from './interfaces'

// Assets
import AppImage from '@/assets/images/app.png'
import LoginImage from '@/assets/images/auth/login.png'
import RegisterImage from '@/assets/images/auth/register.png'

const AppLayoutAuth = memo(({ isLogin }: IAppLayoutAuthProps) => {
  // Hook
  const { t } = useTranslation()

  return (
    <StyledWrapper>
      <StyledContainer>
        <StyledFormContainer>
          {/* Header & Icon - Left Side */}
          <div
            className='flex flex-col items-center gap-8 mb-4'
            data-testid='auth-layout-left-side'
          >
            <img src={AppImage} alt='App' width={'90px'} height={'55.5px'} />

            <AppBaseLabel isBold fontSize={26}>
              {t(`auth.${isLogin ? 'login' : 'register'}`)}
            </AppBaseLabel>
          </div>

          {/* Content Here */}
          <AppRouteWrapper />
        </StyledFormContainer>

        {/* Banner - Right Side */}
        <StyledBanner data-testid='auth-layout-right-side'>
          <img
            src={isLogin ? LoginImage : RegisterImage}
            alt='App'
            width={'647px'}
            height={'602px'}
          />
        </StyledBanner>
      </StyledContainer>
    </StyledWrapper>
  )
})

AppLayoutAuth.displayName = 'AppLayoutAuth'

export { AppLayoutAuth }
