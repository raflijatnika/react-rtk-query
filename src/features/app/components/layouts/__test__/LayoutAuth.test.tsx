// Test
import { render, screen } from '@testing-library/react'
import 'jest-styled-components'

// Components
import { AppLayoutAuth } from '../AppLayoutAuth'

it('AppLayoutAuth should be rendered', () => {
  const { container } = render(
    <AppLayoutAuth>
      <h1 data-testid='content'>Content</h1>
    </AppLayoutAuth>
  )
  expect(container.innerHTML).toMatchSnapshot()
  expect(screen.getByTestId('content').textContent).toEqual('Content')
})

it('AppLayoutAuth should be have optional props isLogin', () => {
  render(
    <AppLayoutAuth data-testid='layout-auth' isLogin>
      <h1 data-testid='content'>Content</h1>
    </AppLayoutAuth>
  )
})

it('AppLayoutAuth left side should be have 2 content statically', () => {
  render(
    <AppLayoutAuth>
      <h1 data-testid='content'>Content</h1>
    </AppLayoutAuth>
  )
  const element = screen.getByTestId('auth-layout-left-side')
  expect(element.childElementCount).toEqual(2)
})

it('AppLayoutAuth left side should be have 1 content statically', () => {
  render(
    <AppLayoutAuth>
      <h1 data-testid='content'>Content</h1>
    </AppLayoutAuth>
  )
  const element = screen.getByTestId('auth-layout-right-side')
  expect(element.childElementCount).toEqual(1)
})
