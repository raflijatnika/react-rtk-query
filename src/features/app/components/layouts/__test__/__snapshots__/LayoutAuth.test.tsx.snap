// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutAuth should be rendered 1`] = `"<div class=\\"sc-bczRLJ kJglZm\\"><div class=\\"sc-gsnTZi SmaeV\\"><div class=\\"sc-dkzDqf dLuZiL\\"><div class=\\"flex flex-col items-center gap-8 mb-4\\" data-testid=\\"auth-layout-left-side\\"><img src=\\"app.png\\" alt=\\"App\\" width=\\"90px\\" height=\\"55.5px\\"><label font-size=\\"26\\" class=\\"sc-eCYdqJ hsohzK\\">auth.register</label></div><h1 data-testid=\\"content\\">Content</h1></div><div data-testid=\\"auth-layout-right-side\\" class=\\"sc-hKMtZM gxGoaX\\"><img src=\\"register.png\\" alt=\\"App\\" width=\\"647px\\" height=\\"602px\\"></div></div></div>"`;
